packages/rss-handler/src/index.ts

```diff
@@ -185,162 +220,182 @@ const getLink = (links: RssFeedItemLink[]) => {
+    console.log('feed has not been updated', feedUrl, lastFetchedChecksum)
+    return
+  }
+  const updatedLastFetchedChecksum = fetchResult.checksum
+
+  // fetch feed
+  let itemCount = 0
+  const feed = await parser.parseString(fetchResult.content)
+  console.log('Fetched feed', feed.title, new Date())
+
+  const feedPubDate = (feed['dc:date'] ||
+    feed.pubDate ||
+    feed.lastBuildDate) as string | undefined
+  console.log('Feed pub date', feedPubDate)
+  if (feedPubDate && new Date(feedPubDate) < new Date(lastFetchedAt)) {
+    console.log('Skipping old feed', feedPubDate)
+    return
+  }
+
+  // save each item in the feed
+  for (const item of feed.items) {
+    // use published or updated if isoDate is not available for atom feeds
+    item.isoDate =
+      item.isoDate ||
+      (item.published as string) ||
+      (item.updated as string) ||
+      (item.created as string)
+    console.log('Processing feed item', item.links, item.isoDate)
+
+    if (!item.links || item.links.length === 0) {
+      console.log('Invalid feed item', item)
+      continue
     }
 
-    const token = req.header('Omnivore-Authorization')
-    if (!token) {
-      console.error('Missing authorization header')
-      return res.status(401).send('UNAUTHORIZED')
+    item.link = getLink(item.links as RssFeedItemLink[])
+    if (!item.link) {
+      console.log('Invalid feed item links', item.links)
+      continue
     }
 
-    try {
-      let userId: string
+    console.log('Fetching feed item', item.link)
 
-      try {
-        const decoded = jwt.verify(token, process.env.JWT_SECRET) as {
-          uid: string
-        }
-        userId = decoded.uid
-      } catch (e) {
-        console.error('Authorization error', e)
-        return res.status(401).send('UNAUTHORIZED')
-      }
+    const publishedAt = item.isoDate ? new Date(item.isoDate) : new Date()
+    // remember the last valid item
+    if (
+      !lastValidItem ||
+      (lastValidItem.isoDate && publishedAt > new Date(lastValidItem.isoDate))
+    ) {
+      lastValidItem = item
+    }
 
-      if (!isRssFeedRequest(req.body)) {
-        console.error('Invalid request body', req.body)
-        return res.status(400).send('INVALID_REQUEST_BODY')
-      }
+    // Max limit per-feed update
+    if (itemCount > 99) {
+      continue
+    }
 
-      const { feedUrl, subscriptionId, lastFetchedAt, lastFetchedChecksum } =
-        req.body
-      console.log('Processing feed', feedUrl, lastFetchedAt)
+    // skip old items and items that were published before 24h
+    if (
+      publishedAt < new Date(lastFetchedAt) ||
+      publishedAt < new Date(Date.now() - 24 * 60 * 60 * 1000)
+    ) {
+      console.log('Skipping old feed item', item.link)
+      continue
+    }
 
-      let lastItemFetchedAt: Date | null = null
-      let lastValidItem: Item | null = null
+    const created = await createSavingItemTask(userId, feedUrl, item)
+    if (!created) {
```

Comment:
I noticed in the other spot this fails we throw an exception, but here we continue. I guess because need to keep processing the other items